### PR TITLE
GEODE-7875:  fix create index gfsh command on partitioned region

### PR DESCRIPTION
**Pull request by mkevo**
The new solution is added regarding comment from the [dev list](https://markmail.org/message/uxhlysb2nx3u7p6y).
When creating an index on the partition region it should just send to one server as it will propagate the create index message to the rest of the members of that group(cluster or one the user specified in the command).

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Comment by jinmeiliao** [comment]
i like the idea of fixing this command for partition regions. The problem does exist that for partition region, the index creation is distributed to all the other servers, so subsequent `CreateIndexFunction` might get name conflict error. We can solve this by either of the two ways:
1) In the `CreateIndexCommand`, if user is creating the index on a partition region, only send the function to one member, not all members. Use `GfshCommand.findAnyMembersForRegion` method to get just one member to execute the function.
2) In the `CreateIndexFunction`,  when we get a `NameConflictException` or `IndexExistsException`, make the `CliFunctionResult` with a `StatusState.IGNORABLE`, this will make the command not consider this as a failure.

Either of these two approaches should work.

**Review by jinmeiliao — CHANGES_REQUESTED** [review]
As above

File: geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java
Comment (jinmeiliao) [inline geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java:98]: I would recommend putting the determination of member for the partition region here.

**Comment by jinmeiliao** [comment]
If we take the first approach, will the gfsh output only show that the index is created on one member or all members? If it only shows one member, it probably will give the user the wrong idea. Probably the 2nd approach is better.

**Comment by mkevo** [comment]
> If we take the first approach, will the gfsh output only show that the index is created on one member or all members? If it only shows one member, it probably will give the user the wrong idea. Probably the 2nd approach is better.

Yes, I agree with you that with first approach it will show one member, which will give the user the wrong idea. 
I will go with the second option which will allow the command to pass and update the cluster config, and we will have an updated cluster config.

**Comment by mkevo** [comment]
Hi @mhansonp, @kirklund,
can you please check the file you are codeowner?

File: geode-dunit/src/main/java/org/apache/geode/management/internal/cli/commands/IndexCommandsIntegrationTestBase.java
Comment (mhansonp) [inline geode-dunit/src/main/java/org/apache/geode/management/internal/cli/commands/IndexCommandsIntegrationTestBase.java:140]: Why would the status be success if there was a duplicate?

File: geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java
Comment (mhansonp) [inline geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java:100]: Super small nitpick, please combine weird comment lines like this.

File: geode-dunit/src/main/java/org/apache/geode/management/internal/cli/commands/IndexCommandsIntegrationTestBase.java
Comment (mhansonp) [inline geode-dunit/src/main/java/org/apache/geode/management/internal/cli/commands/IndexCommandsIntegrationTestBase.java:175]: Same as above, from a high level this seems like an error.

**Review by mhansonp — COMMENTED** [review]
The error codes are a problem for me. Can you explain why they look reversed?

File: geode-dunit/src/main/java/org/apache/geode/management/internal/cli/commands/IndexCommandsIntegrationTestBase.java
Comment (mkevo) [reply geode-dunit/src/main/java/org/apache/geode/management/internal/cli/commands/IndexCommandsIntegrationTestBase.java:140]: As we agreed in the above comments, using Status.IGNORABLE for some type of the exceptions will solve the problem, the user will get the correct output, that index is successfully created on all members, and the status command is a success.
In the output, if an exception occurs it will print in the status section of the command output IGNORED. Why is this ok?
The index will be created on that servers too, but the create request comes from other servers, and if the request from the locator(from the command) comes later it will say that the index already exists and the command will be a success(before this change the command failed, but index created on all members and cluster config is not updated).
Another thing is when the user runs the create index command multiple times, the command will be successful, but in the printout, the user will see that the index already exists in members, and the status is IGNORED.

**Comment by kirklund** [comment]
@mkevo Hi Mario, Can you please take a look at [GEODE-9089](https://issues.apache.org/jira/browse/GEODE-9089)? It's been failing almost daily for over a year and it started failing right after merging PR #[6008](https://github.com/apache/geode/pull/6008). Thanks!

**Review by kirklund — COMMENTED** [review]
The general idea of the ticket seems like a good idea. I'm not sure about the implementation though, especially to make it look like creating a duplicate index is ok. I think that part is wrong and you should focus on a) check if the index is compatible with how it's currently defined on some of the servers, b) send the index creation request to just the servers that don't have that index.

I would basically defer to Barry since he filed GEODE-7875. I recommend making sure that the implementation matches what he expects and then I'm happy to approve. This is one example in which a non-codeowner really needs to sign off on the PR.

**Comment by mkevo** [comment]
> 

Hi @kirklund,

On [GEODE-9089](https://issues.apache.org/jira/browse/GEODE-9089) I saw that the latest failure was before 2 days on the support/1.14 branch. Also, the failure before was one year ago. 
When I check the support/1.14 branch it doesn't have my fix as it will be delivered as a part of the 1.15.0 release.

**Review by demery-pivotal — APPROVED** [review]

**Pull request merged**
